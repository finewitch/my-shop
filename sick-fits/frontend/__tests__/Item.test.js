import ItemComponent from '../components/Item';
import {shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

const fakeItem = {
    id: 'ABC',
    title: 'cool item',
    description: 'uhuh',
    image: 'dog.jpg',
    largeImage: 'dog1.jpg',
    negotiable: true,
    price: 4000
}

// describe('<Item/>', ()=>{
//     it('renders and displays properly', ()=>{
//         const wrapper = shallow(<ItemComponent item={fakeItem}/>);
//         console.log(wrapper.find('.price').text());
//         expect(wrapper.find('.price').text()).toBe('$30');
//     })
// })  

describe('<Item/>', ()=>{
    it('renders and matches snapshot', ()=>{
        // const price = '$50,34'; 
        // expect(price).toMatchSnapshot(); 
        const wrapper = shallow(<ItemComponent item={fakeItem}/>);
        expect(toJSON(wrapper)).toMatchSnapshot();
    })
})