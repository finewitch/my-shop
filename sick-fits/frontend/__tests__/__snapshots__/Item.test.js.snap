// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> renders and matches snapshot 1`] = `
<div
  className="p_wrapper"
>
  <span
    className=" p_wrapper-num c-w"
  />
  <div
    className="p_wrapper-inner"
  >
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "ABC",
          },
        }
      }
    >
      <div
        className="p_wrapper-inner-w"
      >
        <span>
          <img
            alt="cool item"
            src="dog.jpg"
          />
        </span>
        <span>
          <div
            className="c-w price"
          >
            $40
          </div>
          <div
            className="c-w"
          >
            cool item
          </div>
          <div
            className="c-w"
          >
            <div>
              do negocjacji
            </div>
          </div>
          <div
            className="c-w"
          >
             by author
          </div>
        </span>
      </div>
    </Link>
    <div
      className="p_wrapper-inner-btnlist"
    >
      <Link
        href={
          Object {
            "pathname": "update",
            "query": Object {
              "id": "ABC",
            },
          }
        }
      >
        <button>
          edit
        </button>
      </Link>
      <AddToCart
        id="ABC"
      />
      <DeleteItem
        id="ABC"
      />
    </div>
  </div>
</div>
`;
